builder/builder.go

```diff
@@ -217,11 +217,19 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 		// to accomodate parallelism in the WAIT/END PopWaitBlock handling
 		dirIDs = map[int]string{}
 	)
+
 	var (
 		depIndex   = 0
 		imageIndex = 0
 		dirIndex   = 0
 	)
+
+	// Delay closing the registry proxy server until this method
+	// returns. Previously, it was being closed when the buildFunc returned,
+	// which led to a problem with fetching some images.
+	stopRegistryProxyFunc := func() {}
+	defer stopRegistryProxyFunc()
```

Comment:
I don't think this works, see the following for an example:

```
alex@surfperch:~$ cat main.go 
package main

import "fmt"

func main() {

	stopRegistryProxyFunc := func() {
		fmt.Println("first func")
	}
	defer stopRegistryProxyFunc()

	stopRegistryProxyFunc = func() {
		fmt.Println("is this actually called?")
	}
}
alex@surfperch:~$ go run main.go 
first func
```

Reply:
Good point. I was originally using the below approach but wanted to condense the code. It works, but this is because the registry never closes.

```golang
	var stopRegistryProxyFunc func()
	defer func() {
		if stopRegistryProxyFunc != nil {
			stopRegistryProxyFunc()
		}
	}()
```

Reply:
`defer func() { stopRegistryProxyFunc() }` would probably work.

Reply:
Sure, that's fine too.